man/systemd-system.conf.xml

```diff
@@ -361,8 +361,10 @@
         <listitem><para>Configure the default value for the per-unit <varname>TasksMax=</varname> setting. See
         <citerefentry><refentrytitle>systemd.resource-control</refentrytitle><manvolnum>5</manvolnum></citerefentry>
         for details. This setting applies to all unit types that support resource control settings, with the exception
-        of slice units. Defaults to 15%, which equals 4915 with the kernel's defaults on the host, but might be smaller
-        in OS containers.</para></listitem>
+        of slice units. Defaults to 15% of the kernel's option <varname>kernel.pid_max=</varname> or root cgroup <varname>pids.max</varname>.
+        Kernel has a default value for <varname>kernel.pid_max=</varname> and an algorithm of counting in case of more than 32 cores.
+        For example with default <varname>kernel.pid_max=</varname>, <varname>DefaultTasksMax=</varname> will be equal to 4915,
+        but might be greater in other systems or smaller in OS containers.</para></listitem>
```

Comment:
Are there always greater or equal (not smaller) in non-container systems?

Reply:
Yes, links to the kernel code attached. The default value for `kernel.pid_max=` is 32768, so 15% is 4915 and for systems with 32+ cores the formula is 1024 * NUM_CORES.